perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -852,29 +852,54 @@ private void setFromAndReplyTo(MimeMessage message, String fromName, String from
 
 	/**
 	 * Get proper values "TO" for mail message based on VO or GROUP attribute "toEmail"
-	 *
+	 * <p>
 	 * If group attribute not set and is group application, get vo attribute as backup.
 	 * If no attribute set, BACKUP_FROM address will be used.
 	 *
 	 * @param app application to decide if it's VO or Group application
-	 *
 	 * @return list of mail addresses to send mail to
 	 */
-	private List<String> getToMailAddresses(Application app){
+	private List<String> getToMailAddresses(Application app) {
 		List<String> result = new ArrayList<>();
 
 		// get proper value from attribute
 		try {
-			Attribute attrToEmail = getMailFromVoAndGroupAttrs(app, URN_VO_TO_EMAIL, URN_GROUP_TO_EMAIL);
+			Attribute attrSenderEmail;
+
+			if (app.getGroup() == null) {
+				// it is a VO application
+				return getMailsFromVo(app, URN_VO_TO_EMAIL);
+			}
+
+			attrSenderEmail = attrManager.getAttribute(registrarSession, app.getGroup(), URN_GROUP_TO_EMAIL);
+
+			if (attrSenderEmail == null || attrSenderEmail.getValue() == null) {
+				// not specified toEmail group attribute
+
+				// try to use all group admins with specified preferred emails
+				List<User> admins = groupsManager.getAdmins(registrarSession, app.getGroup(), true);
+				for (User admin : admins) {
+					attrSenderEmail = attrManager.getAttribute(registrarSession, admin, URN_USER_PREFERRED_MAIL);
 
-			if (attrToEmail != null && attrToEmail.getValue() != null) {
-				ArrayList<String> value = attrToEmail.valueAsList();
-				for (String adr : value) {
-					if (adr != null && !adr.isEmpty()) {
-						result.add(adr);
+					if (attrSenderEmail != null && attrSenderEmail.getValue() != null) {
+						result.add(BeansUtils.attributeValueToString(attrSenderEmail));
```

Comment:
Please, use this instead.
```suggestion
						result.add(attrSenderEmail.valueAsString());
```